packages/backend/src/execution/executors/PgExecutor/PgExecutor.ts

```diff
@@ -43,20 +42,29 @@ export class PgExecutor implements QueryExecutor<PgQueryResult> {
 
     async execute(
         query: AnyQuery,
-        { defaultSchema }: { defaultSchema: string },
+        {
+            defaultSchema,
+            prependSql,
+        }: { defaultSchema: string; prependSql?: string },
     ): Promise<Array<PgQueryResult>> {
         const client = await this.client;
+
         const { sqlBuilder, augmentedQuery } = composeQuery({
             defaultSchema,
             query,
         });
+
         const { params, sql } = sqlBuilder.build();
 
         try {
             if (this.props.logging) {
                 console.log(format(sql, { language: 'postgresql' }));
             }
 
+            if (prependSql) {
+                await client.query(prependSql);
```

Comment:
that's not right: we shouldn't open two SQL queries, we want to concatenate the queries so they are part of the same transaction.

Reply:
So we want something like this:

```
begin
{prependSql};
{sql};
commit
```

Reply:
So this would look like this for Luminovo's case:

```ts
const transformSql = (sql:string) => {
    return `begin set "my.tenant"=`someTenant`; ${sql}; commit;`
}
queryEngine.execute(query,{transformSql})
```